include/libtorrent/torrent_handle.hpp

```diff
@@ -873,9 +871,12 @@ namespace libtorrent
 			, std::string const& private_key
 			, std::string const& dh_params);
 
+#ifndef TORRENT_NO_DEPRECATE
 		// Returns the storage implementation for this torrent. This depends on the
 		// storage constructor function that was passed to add_torrent.
+		TORRENT_DEPRECATED
 		storage_interface* get_storage_impl() const;
```

Comment:
this is a bit trickier I think. The reason is that libtorrent creates and owns the storage object, and even though it was user-supplied, the user doesn't have a reasonable way to talk to it (other than this function). The alternative (with the current design) would be for the storage_constructor object to keep a reference to an outside pointer, and fill that in when it creates a storage object. Another alternative would be for the user to construct the storage up-front, and pass in the storage object via add_torrent_params, rather than the storage_constructor function. However, that makes the storage object a bit messy, since it can't be properly initialized by the user, it would require two-phase initialization.